### ISSUE TITLE
[BUG] Dependency custom rule is not triggered if credencial is not attached

### ISSUE THREAD

**Issue by mguetta1** [konveyor/analyzer-lsp#884]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

0.8.0-alpha.4

### Priority

Major

### Current Behavior

Dependency custom rule is not triggered if credencial is not attached to the application

### Expected Behavior

The custom rule should be triggered regardless existence of credencials

### How Reproducible

Always (Default)

### Steps To Reproduce

Analyze [tackle-test-public](https://github.com/konveyor/tackle-testapp-public) app with the following analysis configuration:
1. Source code + dependencies
2. Skip "Set targets"
3. Upload [this](https://gist.github.com/mguetta1/2805e72a09c7bd090e9a9ba91c80e5f9) as manual rules file
4. No additional advanced options

### Environment

```markdown

```
